### PR TITLE
#133 add link varian with icon

### PR DESCRIPTION
**Pull request by mablancoalvarez**
@griselmatosm and I created a new temporary icon called "ExternalLink" and we added it to Button component just to testing , We have doubts about how we will use it in the rest of the components where links is includes.

### CODE REVIEW

File: src/Icon/index.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ import Search from './Search'
 import Success from './Success'
 import User from './User'
 import Warning from './Warning'
-import ExternalLink from './ExternalLink'
+import IconLink from './IconLink'
```
Comment (oscarotero) [inline src/Icon/index.js:24]: I the icon should named just `Link`, instead `IconLink` (the other icons do not include the term `Icon` in their names).

File: src/Button/index.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ Button.defaultProps = {
   busy: null,
   className: null,
   disabled: false,
-  iconLink : false
+  iconLink: false
```
Comment (oscarotero) [inline src/Button/index.js:35]: The attribute should be just `icon`, instead `iconLink`.  Why only links can have icons? Let's make this functionality more generical.

File: src/Button/index.js [hunk 3]
```diff
@@ -11,17 +11,17 @@ const Button = ({ variant, children, disabled, busy, className, iconLink, as, ..
...
+    { 'is-disabled': disabled && !busy },
     className
   )
   const isIcon = (variant === 'link' && iconLink)
   return (
     <ElementType className={styles} disabled={busy ? true : disabled} {...props}>
       {busy && <Spinner />}
       <span className="ola_button-text">{!busy ? children : busy}</span>
-      {isIcon && <Icon name="externalLink"/> }
+      {isIcon && <Icon name="iconLink" />}
```
Comment (oscarotero) [inline src/Button/index.js:24]: We should be able to insert any icon, not only for links. The property should accept a string with the icon name instead a boolean.

File: src/Button/index.js [hunk 4]
```diff
@@ -56,7 +56,7 @@ Button.propTypes = {
   /** Button is disabled */
   disabled: PT.bool,
   /**Icon is a Bolean  */
-  iconLink: PT.bool
+  icon: PT.string
```
Comment (griselmatosm) [inline src/Button/index.js:59]: The icon prop is now a string

File: src/Button/stories.js [hunk 5]
```diff
@@ -20,7 +20,7 @@ storiesOf('Button')
       <figure><Button href="#" variant="primary" target="_blank">Link</Button></figure>
       <figure><Button href="#" variant="primary" disabled>Link disabled</Button></figure>
       <figure><Button href="#" variant="primary" busy='Link busy ...' disabled>Hide text by busy text</Button></figure>
-      <figure><Button href="#" variant="link" iconLink={true}>Link style</Button></figure>
+      <figure><Button href="#" variant="link" icon="success">Link style</Button></figure>
```
Comment (griselmatosm) [inline src/Button/stories.js:23]: Here we have tested with another existing icons to check this is reusable

File: src/Icon/index.js [hunk 6]
```diff
@@ -43,7 +43,7 @@ const ICONS = {
   'success': Success,
   'user': User,
   'warning': Warning,
-  'iconLink': IconLink
+  'link': Link
```
Comment (griselmatosm) [inline src/Icon/index.js:46]: change names

File: src/Tag/stories.js [hunk 7]
```diff
@@ -15,17 +13,18 @@ storiesOf('Tag')
...
     </figure>
   ))
   .add('With a ButtonIcon', () => (
     <figure>
       <Tag
         size={radios('Size', ['small', 'medium', 'big'], 'small')}
-        variant={radios('Variant', ['invert', 'error', 'pro', 'success', 'warning'])}>
-        <span>{ text('Label', 'This is a tag') }</span>
-        <ButtonIcon><Icon size="small" name="close" /></ButtonIcon>
+        variant={radios('Variant', ['invert', 'error', 'pro', 'success', 'warning'])}
```
Comment (griselmatosm) [inline src/Tag/stories.js:24]: We have a proposal to use the icon props in another component that use an icon. The goal of this is testing if is possible to use that prop and it's working fine.

File: src/Tag/stories.js [hunk 8]
```diff
@@ -1,9 +1,11 @@
 import React from 'react'
 import { storiesOf } from '@storybook/react'
 import { text, radios } from '@storybook/addon-knobs'
-
```
Comment (mablancoalvarez) [inline src/Tag/stories.js:4]: We have keep tag component just as it was before

File: src/Icon/index.js [hunk 9]
```diff
@@ -83,7 +83,7 @@ Icon.propTypes = {
     'success',
     'user',
     'warning',
-    'Link'
+    'link'
   ]),
   /** Size */
   size: PT.oneOf(['small', 'medium', 'big']),
```
Comment (mablancoalvarez) [inline src/Icon/index.js:89]: add icon into a viewer

File: src/Button/stories.js [hunk 10]
```diff
@@ -20,7 +20,7 @@ storiesOf('Button')
       <figure><Button href="#" variant="primary" target="_blank">Link</Button></figure>
       <figure><Button href="#" variant="primary" disabled>Link disabled</Button></figure>
       <figure><Button href="#" variant="primary" busy='Link busy ...' disabled>Hide text by busy text</Button></figure>
-      <figure><Button href="#" variant="link" icon="success">Link style</Button></figure>
+      <figure><Button href="#" variant="link" icon="link">Link style</Button></figure>
     </div>
```
Comment (mablancoalvarez) [inline src/Button/stories.js:24]: we have change de icon to a "link" so now you could see it into a button in the storybook

**Pull request merged**